### ISSUE TITLE
Improve the reporting of invalid input values

### ISSUE THREAD

**Issue by wzxiong** [jpmml/jpmml-evaluator#145]
When convert lightgbm to pmml, there exist error like:
expected: 'DOUBLE', got: '4.0'. error: org.jpmml.evaluator.InvalidResultException

this happen because the pmml file have structure like 

\<DataField name="feature_name" optype="continuous" dataType="double"\>
			\<Interval closure="closedClosed" leftMargin="0.0" rightMargin="3.0"/\>
		\</DataField\>

just remove this part \<Interval closure="closedClosed" leftMargin="0.0" rightMargin="7034.5"/\>, then it will work fine

**Comment by vruusmann** [comment]
> `<Interval closure="closedClosed" leftMargin="0.0" rightMargin="3.0"/>`

This `Interval` element declaration means: "If the value of "feature_name" input field is greater or equal than 0 and smaller or equal than 3.0, then this input value is valid and it's OK to proceed with model scoring. Otherwise, this input value is invalid, and the model cannot be scored".

See section "Values and Intervals" here:
http://dmg.org/pmml/v4-3/DataDictionary.html#xsdElement_Interval

And if you think that this restriction is not appropriate for the particular application scenario, then you should:
1. Extended the bounds of the `Interval` element so that all possible input values would be classified as valid.
2. Delete the `Interval` element.
3. Keep the `Interval` element, but specify "it's OK to proceed with invalid input values" by setting the value of the corresponding `MiningField@invalidValueTreatment` attribute to `"asIs"` (http://dmg.org/pmml/v4-3/MiningSchema.html#xsdType_INVALID-VALUE-TREATMENT-METHOD).

TLDR: The JPMML-Evaluator library is behaving 100% correctly, by preventing you from using a model with invalid input values.

**Comment by vruusmann** [comment]
My takeaway is that perhaps there should be a new, special-purpose exception type for capturing this situation.

The name of the current exception type reads as `InvalidResultException`, which is kind of difficult to relate to the fact that the root cause of the problem is an invalid input value.

**Comment by wzxiong** [comment]
thank you so much. maybe it's not that often to see unseen data on oot dataset
